Docker first look with openssl, mTLS and agent self registration (#158)
neo4j/docs-ops-manager: modules/ROOT/pages/addition/kubernetes/sidecar.adoc

= NOM agent as a Sidecar

> A NOM agent can be deployed as a Sidecar container within the same pod as Neo4j instance in a Kubernetes cluster. Agent sidecar does not need to use an official agent image.

Reviewer: ```suggestion
A NOM agent can be deployed as a Sidecar container within the same pod as Neo4j instance in a Kubernetes cluster. 
The agent Sidecar does not need to use an official agent image.
```

> An agent can simply be run in a lightweight linux based container such as `alpine`. The agent binary can be either mounted into container or can be extracted from Neo4j image using an init container.

Reviewer: ```suggestion
An agent can simply be run in a lightweight Linux-based container such as `alpine`. 
The agent binary can be either mounted into a container or can be extracted from the Neo4j image using an init container.
```

[WARNING]
====

> An agent Sidecar container can not access all resources that Neo4j pod uses and the host metrics of the cluster node. For example, for disk usage metrics, agent needs access to file system or peristent volumes that Neo4j pod uses. These need to be mounted to agent Daemon pod. Agent sidecar can only report host metrics of its own contained environment which is not a complete view of resource usage of cluster node.

Reviewer: ```suggestion
An agent Sidecar container cannot access all resources that the Neo4j pod uses and the host-metrics of the cluster node. 
For example, for disk usage metrics, the agent needs access to the file system or persistent volumes that the Neo4j pod uses. 
These need to be mounted to the agent Daemon pod. 
The agent Sidecar can only report host metrics of its own contained environment which is not a complete view of the resource usage of the cluster node.
```
